### PR TITLE
support of comments in snippet

### PR DESCRIPTION
**Pull request by pvojtechovsky**
The comments in snippet was always ignored. This PR includes this:

- Test case which fails when comments are lost in snippet
- Implementation of comments in snippet
- new method `CompilationUnit JDTSnippetCompiler.getSnippetCompilationUnit()`, which returns CompilationUnit of the compilet snippet. Note: I had to remember the CompilationUnit of the snippet in the CompilationUnit cache, otherwise JDTCommentBuilder was not able to add comments. And at the end I removed that CU from the cache ... so it is now possible to access that CU from the JDTSnippetCompiler too.

It is the first chunk of changes suggested by PR #929

### CODE REVIEW

File: src/main/java/spoon/reflect/factory/CompilationUnitFactory.java [hunk 1]
```diff
@@ -76,6 +79,16 @@ public CompilationUnit create(String filePath) {
 	}
 
 	/**
+	 * Removes compilation unit from the cache and returns it
+	 * Used by JDTSnippetCompiler to remove processed snippet from the cache
+	 * @param filePath
+	 * @return a cached compilation unit or null
+	 */
+	public CompilationUnit remove(String filePath) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/CompilationUnitFactory.java]: rename method to removeFromCache?

Comment (monperrus) [inline src/main/java/spoon/reflect/factory/CompilationUnitFactory.java]: package-visible instead of public?

Comment (pvojtechovsky) [inline src/main/java/spoon/reflect/factory/CompilationUnitFactory.java]: > removeFromCache

good idea

> package-visible

It is different package ... or I do not know this java feature...

**Comment by pvojtechovsky** [comment]
I have committed the rename of method remove->removeFromCache. I do not know how to implement package visibility if the caller is in different package. I think it is not possible. But Travis does not want to check my last commit. OR am I wrong?

**Comment by monperrus** [comment]
thanks!

**Pull request merged**